tests/import_specifications/test_file_parser.py

```diff
@@ -262,3 +264,199 @@ def parseResults_init_fail(
+
+def _get_mocks(count: int) -> tuple[Mock]:
+    return (Mock() for _ in range(count))
+
+
+def test_parse_import_specifications_success():
+    resolver, logger, parser1, parser2 = _get_mocks(4)
+
+    resolver.side_effect = [_ftr(parser1), _ftr(parser2)]
+
+    parser1.return_value = ParseResults(frozendict(
+        {"type1": ParseResult(
+            spcsrc("myfile.xlsx", "tab1"),
+            (frozendict({"foo": "bar"}), frozendict({"baz": "bat"}))
+            ),
+         "type2": ParseResult(
+            spcsrc("myfile.xlsx", "tab2"),
+            (frozendict({"whee": "whoo"}),)  # tuple!
+            )
+        }
+    ))
+
+    parser2.return_value = ParseResults(frozendict(
+        {"type_other": ParseResult(
+            spcsrc("somefile.csv"),
+            (frozendict({"foo": "bar2"}), frozendict({"baz": "bat2"}))
+            )
+        }
+    ))
+
+    res = parse_import_specifications(
+        (Path("myfile.xlsx"), Path("somefile.csv")), resolver, logger
+    )
+
+    assert res == ParseResults(frozendict({
+        "type1": ParseResult(
+            spcsrc("myfile.xlsx", "tab1"),
+            (frozendict({"foo": "bar"}), frozendict({"baz": "bat"}))
+            ),
+        "type2": ParseResult(
+            spcsrc("myfile.xlsx", "tab2"),
+            (frozendict({"whee": "whoo"}),)  # tuple!
+            ),
+        "type_other": ParseResult(
+            spcsrc("somefile.csv"),
+            (frozendict({"foo": "bar2"}), frozendict({"baz": "bat2"}))
+            )
+    }))
+
+    resolver.assert_has_calls([call(Path("myfile.xlsx")), call(Path("somefile.csv"))])
+    parser1.assert_called_once_with(Path("myfile.xlsx"))
+    parser2.assert_called_once_with(Path("somefile.csv"))
+    logger.assert_not_called()
+
+
+def test_parse_import_specifications_fail_no_paths():
+    res = parse_import_specifications(tuple(), lambda p: None, lambda e: None)
+    assert res == ParseResults(errors=tuple([Error(ErrorType.NO_FILES_PROVIDED)]))
+
+
+def test_parse_import_specification_resolver_exception():
+    """
+    This tests an "oh shit" error scenario where we get a completely unexpected error that
+    gets caught in the top level catch block and we bail out.
+    """
+    resolver, logger, parser1 = _get_mocks(3)
+
+    resolver.side_effect = [_ftr(parser1), ArithmeticError("crapsticks")]
+
+    # test that other errors aren't included in the result
+    parser1.return_value = ParseResults(errors=tuple([Error(ErrorType.OTHER, 'foo')]))
+
+    res = parse_import_specifications(
+        (Path("myfile.xlsx"), Path("somefile.csv")), resolver, logger
+    )
+
+    assert res == ParseResults(errors=tuple([Error(ErrorType.OTHER, "crapsticks")]))
+
+    resolver.assert_has_calls([call(Path("myfile.xlsx")), call(Path("somefile.csv"))])
+    parser1.assert_called_once_with(Path("myfile.xlsx"))
+    # In [1]: ArithmeticError("a") == ArithmeticError("a")
+    # Out[1]: False
+    # so assert_called_once_with doesn't work
```

Comment:
Right, the operation is checking to see if the two exceptions are the same instance of the exception. You'd have to check the e.args and type()

You could also `str`ingify the exceptions and then you could use assert_called_once_with, but this assert_exception_correct is probably better.